libs/@local/hashql/core/src/id/bit_vec/mod.rs

```diff
@@ -0,0 +1,2026 @@
+//! This file is derived from the Rust compiler source code.
+//! Source: <https://github.com/rust-lang/rust/blob/f5242367f4acf0130c9b68e6956531b2cb73bd38/compiler/rustc_index/src/bit_set.rs>
+//!
+//! Originally dual-licensed under either of:
+//!   - Apache License, Version 2.0 (see LICENSE-APACHE.md or <https://www.apache.org/licenses/LICENSE-2.0>)
+//!   - MIT license (see LICENSE-MIT.md or <https://opensource.org/licenses/MIT>)
+//!
+//! You may use, copy, modify, and distribute this file under the terms of the
+//! GNU Affero General Public License, Version 3.0, as part of this project,
+//! provided that all original notices are preserved.
+//!
+//! Local adaptations relative to the pinned upstream:
+//! API:
+//! - Use `Id` in place of `Idx`.
+//!   - `index` -> `as_usize`
+//!   - `new` -> `from_usize`
+//! - Mask helpers are no longer generic over `Id` (we do not implement `Id` for `usize`).
+//!
+//! Implementation and maintenance:
+//! - Removed rustc-specific serialization code.
+//! - Migrated to smallvec v2.
+//! - Replaced `unwrap` with `unwrap_or_else(|| unreachable!(...))` where applicable.
+//! - Applied clippy-driven fixes (no intended semantic changes).
+//! - Removed inherent impls for bit relations.
+//! - Removed used of `static_assert_size` macro in favor of `assert_eq!`.
+//! - Removed mentions of `IntervalSet`.
+#![coverage(off)] // Vendored in module with tests
```

Comment:
Yes, this is vendored in, but we have tests for that, so let's keep this on, please. It would be useful to see in codecov if vendored-in code is **not** tested.